File: packages/secure_mnemonic/android/src/main/kotlin/com/adguard/cryptowallet/secure_mnemonic/enums/MethodName.kt
package com.adguard.cryptowallet.secure_mnemonic.enums

enum class MethodName {
    GET_TPM_STATUS,
    GET_BIOMETRY_STATUS,
    GENERATE_KEY,
    ENCRYPT,
    DECRYPT,
    DELETE_KEY,
    CONFIGURE;

    override fun toString(): String =
        when (this) {
            GET_TPM_STATUS -> "getTPMStatus"
            GET_BIOMETRY_STATUS -> "getBiometryStatus"
            GENERATE_KEY -> "generateKey"
            ENCRYPT -> "encrypt"
            DECRYPT -> "decrypt"
            DELETE_KEY -> "deleteKey"
            CONFIGURE -> "configure"
        }
}

File: packages/secure_mnemonic/windows/method_name.cpp
#include "include/secure_mnemonic/enums/method_name.h"

#include <unordered_map>
#include <stdexcept>

using secure_mnemonic::MethodName;

namespace secure_mnemonic {
	const std::unordered_map<std::string, MethodName> METHOD_NAME_MAP = {
		{"getTPMStatus", MethodName::kGetTPMStatus},
		{"getBiometryStatus", MethodName::kGetBiometryStatus},
		{"generateKey", MethodName::kGenerateKey},
		{"encrypt", MethodName::kEncrypt},
		{"decrypt", MethodName::kDecrypt},
		{"deleteKey", MethodName::kDeleteKey},
		{"configure", MethodName::kConfigure},
		{"notImplemented", MethodName::kNotImplemented},
	};

	MethodName GetMethodName(const std::string& methodName)
	{
		auto it = METHOD_NAME_MAP.find(methodName);
		if (it != METHOD_NAME_MAP.end())
		{
			return it->second;
		}
		else {
			return MethodName::kNotImplemented;
		}
	}
}
